ocp_resources/network_attachment_definition.py

```diff
@@ -9,19 +11,147 @@ class NetworkAttachmentDefinition(NamespacedResource):
     api_group = NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO
     resource_name = None
 
-    def wait_for_status(
-        self, status, timeout=None, label_selector=None, resource_version=None
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        bridge_name=None,
+        cni_type=None,
+        vlan=None,
+        client=None,
+        mtu=None,
+        macspoofchk=None,
+        teardown=True,
+        old_nad_format=False,
+        add_resource_name=True,
+        dry_run=None,
     ):
-        raise NotImplementedError(f"{self.kind} does not have status")
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            dry_run=dry_run,
+        )
+        self.old_nad_format = old_nad_format
+        self.bridge_name = bridge_name
+        self.cni_type = cni_type
+        self.vlan = vlan
+        self.mtu = mtu
+        self.macspoofchk = macspoofchk
+        self.add_resource_name = add_resource_name
 
     def to_dict(self):
         res = super().to_dict()
         if self.yaml_file:
             return res
 
+        res["spec"] = {}
         if self.resource_name is not None:
             res["metadata"]["annotations"] = {
                 f"{NamespacedResource.ApiGroup.K8S_V1_CNI_CNCF_IO}/resourceName": self.resource_name
             }
-        res["spec"] = {}
+
+        spec_config = {"cniVersion": "0.3.1", "name": self.bridge_name}
+        bridge_dict = {"type": self.cni_type, "bridge": self.bridge_name}
+        if self.mtu:
+            bridge_dict["mtu"] = self.mtu
+        if self.vlan:
+            bridge_dict["vlan"] = self.vlan
+        if self.old_nad_format:
+            spec_config["plugins"] = [bridge_dict]
+        else:
+            spec_config.update(bridge_dict)
+        if self.macspoofchk:
+            spec_config["macspoofchk"] = self.macspoofchk
+
+        res["spec"]["config"] = spec_config
+        return res
+
+    def wait_for_status(
+        self, status, timeout=None, label_selector=None, resource_version=None
+    ):
+        raise NotImplementedError(f"{self.kind} does not have status")
+
+
+class LinuxBridgeNetworkAttachmentDefinition(NetworkAttachmentDefinition):
+    def __init__(
+        self,
+        name,
+        namespace,
+        bridge_name,
+        cni_type="cnv-bridge",
```

Comment:
Please change the default

Reply:
done

Reply:
removed for ovs
for bridge it can be cnv-bridge" or bridge, leaving the default as it